torch/distributed/rpc/api.py

```diff
@@ -66,10 +89,45 @@ def _wait_all_workers():
     terminate the RPC framework, and there is no guarantee that the RPC
     framework will work after this method returns.
     """
-    global _agent
+    if _agent is None:
+        return
+
+    assert _ALL_WORKER_NAMES is not None, (
```

Comment:
why using names instead of IDs for this?

Reply:
Is it because we would like to have better error message in `_on_master_follower_report_done `?

Reply:
Right, just for ease of debugging. It's not an API on the critical path like send/recv. So I didn't put squeezing performance on my mind. We could always optimize it and use `get_worker_info(id).name` to print debugging messages.